local/php/envs.go

```diff
@@ -31,17 +31,36 @@ import (
 
 func (p *Server) generateEnv(req *http.Request) map[string]string {
 	scriptName := p.passthru
+
 	https := ""
 	if req.TLS != nil {
 		https = "On"
 	}
 
 	pathInfo := req.URL.Path
+	filenameMatch := false
+
 	if pos := strings.Index(strings.ToLower(pathInfo), ".php"); pos != -1 {
 		file := pathInfo[:pos+4]
 		if _, err := os.Stat(filepath.Join(p.documentRoot, file)); err == nil {
 			scriptName = file
 			pathInfo = pathInfo[pos+4:]
+			filenameMatch = true
+		}
+
+	}
+
+	if filenameMatch == false && pathInfo != "/" && len(pathInfo) > 0 {
+		paths := strings.Split(strings.Trim(pathInfo, "/"), "/")
+		for i, n := 0, len(paths); i < n; i++ {
+			poppedPath := paths[:len(paths)-i]
+			indexDir := filepath.Join(poppedPath...)
+			file := filepath.Join(indexDir, p.passthru)
+			if _, err := os.Stat(filepath.Join(p.documentRoot, file)); err == nil {
+				scriptName = string(os.PathSeparator) + file
```

Comment:
`scriptName` is not only used for the file path but also to populate some web-related environments.
So if it's not entirely accurate I would rather stick with a `"/"` prefix to remain consistent with every other passthru usage we have and not break anything in the PHP userland side.

The end result goes through `filepath.Join` which will use `filepath.Clean` so this should not cause any issue on Windows anyway for PHP (engine side this time).